core/src/main/java/io/parsingdata/metal/data/DataExpressionSource.java

```diff
@@ -81,21 +81,21 @@ private Value getValue() {
 
     @Override
     public String toString() {
-        return getClass().getSimpleName() + "(" + dataExpression.toString() + "[" + index + "](" + graph + "," + encoding + "))";
+        return getClass().getSimpleName() + "(" + dataExpression.toString() + "[" + index + "](" + parseState + "," + encoding + "))";
     }
 
     @Override
     public boolean equals(Object obj) {
         return Util.notNullAndSameClass(this, obj)
             && Objects.equals(dataExpression, ((DataExpressionSource)obj).dataExpression)
             && Objects.equals(index, ((DataExpressionSource)obj).index)
-            && Objects.equals(graph, ((DataExpressionSource)obj).graph)
+            && Objects.equals(parseState, ((DataExpressionSource)obj).parseState)
             && Objects.equals(encoding, ((DataExpressionSource)obj).encoding);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(getClass().hashCode(), dataExpression, index, graph, encoding);
+        return Objects.hash(getClass().hashCode(), dataExpression, index, parseState, encoding);
```

Comment:
You can probably remove the calls to hashcode everywhere where using ```Objects#hash```?

Reply:
It seems to get called on it anyway and all tests are green. Good catch!